core/txpool/legacypool/list.go

```diff
@@ -281,9 +280,10 @@ type list struct {
 	strict bool       // Whether nonces are strictly continuous or not
 	txs    *sortedMap // Heap indexed sorted hash map of the transactions
 
-	costcap   *big.Int // Price of the highest costing transaction (reset only if exceeds balance)
```

Comment:
Why don't we need `costcap` any more?

Reply:
Feels like it could be a map like `totalcost`, I need to check if that would allow to simplify the code though.

Reply:
Yes, I've already implemented it, but fixing some pesky bugs. The whole thing about mixing currencies and pointers to currencies has made it a little bit annoying, so I'm trying to keep some coherence in some parts